### PR TITLE
complete job when app container exit in sidecar mode

### LINKED ISSUE #556: [ENHANCE] sidecar: complete the injected pod as expected
Refer to #448 and https://github.com/juicedata/juicefs-csi-driver/pull/454#pullrequestreview-1278629782

### PR DESCRIPTION
**Pull request by zwwhdls**
fix https://github.com/juicedata/juicefs-csi-driver/issues/556

### CODE REVIEW

File: pkg/controller/app_controller.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+}
+
+func (a *AppController) umountFuseSidecar(pod *corev1.Pod, fuseContainer corev1.Container) (err error) {
+	if fuseContainer.Name == "" {
+		return
+	}
+
+	cmd := []string{}
+	// get prestop
+	if fuseContainer.Lifecycle != nil && fuseContainer.Lifecycle.PreStop != nil && fuseContainer.Lifecycle.PreStop.Exec != nil {
```
Comment (Hexilee) [inline pkg/controller/app_controller.go]:
```suggestion
	if fuseContainer.Lifecycle == nil || fuseContainer.Lifecycle.PreStop == nil || fuseContainer.Lifecycle.PreStop.Exec == nil {
	    return nil
	}
```

Comment (Hexilee) [reply pkg/controller/app_controller.go]: To avoid executing empty command

File: pkg/controller/app_controller.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+			strings.Contains(stderr, "no mount point specified") {
+			// if mount point not mounted, do not retry
+			return nil
+		}
+		if strings.Contains(err.Error(), "exit code 137") {
+			klog.Warningf("[AppController] exec with exit code 137, ignore it. error: %v", err)
+			return nil
+		}
+		klog.Errorf("[AppController] error: %v; exec stdout: %s; exec stderr: %s", err, stdout, stderr)
+		return err
```
Comment (Hexilee) [inline pkg/controller/app_controller.go]: Should removed as duplicated with the line 109

File: pkg/controller/app_controller.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+			}
+
+			klog.V(6).Infof("[pod.onCreate] pod [%s] in [%s] requeue", pod.Name, pod.Namespace)
+			return true
+		},
+		UpdateFunc: func(updateEvent event.UpdateEvent) (needUpdate bool) {
+			podNew, ok := updateEvent.ObjectNew.(*corev1.Pod)
+			if !ok {
+				klog.V(6).Infof("[pod.onUpdate] can not turn into pod, Skip. object: %v", updateEvent.ObjectNew)
+				return needUpdate
```
Comment (Hexilee) [inline pkg/controller/app_controller.go]:
```suggestion
				return
```

or 

```suggestion
				return false
```

File: pkg/controller/app_controller.go (hunk 1 above)
Comment (zwwhdls) [reply pkg/controller/app_controller.go]: done

File: pkg/controller/app_controller.go (hunk 2 above)
Comment (zwwhdls) [reply pkg/controller/app_controller.go]: done

File: pkg/controller/app_controller.go (hunk 3 above)
Comment (zwwhdls) [reply pkg/controller/app_controller.go]: done

**Pull request merged**